### DISCUSSION TITLE
Is there a difference between sideEffects being undefined vs. being set to true?

### DISCUSSION THREAD

**Discussion by cosst** [webpack/webpack#16551] [Q&A] [answered]
My assumption was that not explicitly specifying a boolean value for `sideEffects` (IE `undefined`) would be equivalent to setting `"sideEffects": true`, but [this section](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) seems to imply that even with `sideEffects` being undefined there is some automatic omission of unneeded code.

I guess my specific question is: if we start explicitly setting `"sideEffects": true`, will we be losing any automatic-optimization that is happening when we leave it undefined?

**Comment by alexander-akait** [comment, marked as answer]
> My assumption was that not explicitly specifying a boolean value for sideEffects (IE undefined) would be equivalent to setting "sideEffects": true, but [this section](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) seems to imply that even with sideEffects being undefined there is some automatic omission of unneeded code.

No `sideEffects` === `"sideEffects": true`, but yes, we can do tree shaking

> I guess my specific question is: if we start explicitly setting "sideEffects": true, will we be losing any automatic-optimization that is happening when we leave it undefined?

You will have the same behaviour
